code/game/objects/items/food/pizza.dm

```
@@ -16,11 +16,102 @@
+
+/obj/item/food/pizza/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/drag_pickup)
+	register_context()
+
+/obj/item/food/pizza/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+
+	if(istype(held_item) && (held_item.tool_behaviour in cutting_tools))
+		context[SCREENTIP_CONTEXT_LMB] = "Slice"
+		context[SCREENTIP_CONTEXT_RMB] = "Slice apart"
+		return TRUE
+
+	if(sliced)
+		context[SCREENTIP_CONTEXT_LMB] = "Take Slice"
+		return TRUE
+
+/obj/item/food/pizza/examine(mob/user)
+	. = ..()
+	if(isnull(slice_type) || !sliced)
+		return
+	. += span_notice("You can slice this to make it possible to take out slices with an empty hand!")
+
+/obj/item/food/pizza/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	. = NONE
+	if(isnull(slice_type) || !(tool.tool_behaviour in cutting_tools))
+		return
+	if(!sliced)
+		slice(user, tool)
+		return ITEM_INTERACT_SUCCESS
+	user.visible_message(span_notice("[user] seperates [src] into individual slices with [tool]."))
+	cut_apart()
+	return ITEM_INTERACT_SUCCESS
+
+/obj/item/food/pizza/item_interaction_secondary(mob/living/user, obj/item/tool, list/modifiers)
+	. = NONE
+	if(isnull(slice_type) || !(tool.tool_behaviour in cutting_tools))
+		return
+	visible_message(span_notice("[user] seperates [src] into individual slices with [tool]."))
+	cut_apart()
+	return ITEM_INTERACT_SUCCESS
+
+/obj/item/food/pizza/attack_hand(mob/user, list/modifiers)
+	. = ..()
+	if(!sliced)
+		return
+	user.visible_message(span_notice("You take a slice of [src]."), span_notice("[user] takes a slice of [src]."))
+	produce_slice(user)
+
+/obj/item/food/pizza/proc/get_slices_filter() //to not repeat code
+	return alpha_mask_filter(icon = icon('icons/obj/food/pizza.dmi', "[slices_left]slices"))
 
+/// makes us possible to take slices out of
+/obj/item/food/pizza/proc/slice(mob/user, obj/item/tool)
+	if(sliced)
+		return
+	tool?.play_tool_sound(src)
```

We should error if called without a tool not silently ignore

---

pizza boxes arrive presliced